runbotics/runbotics-orchestrator-ui/src/src-app/components/ProcessVariablesPanel/VariablesPanel/VariablesPanel.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+
+
+
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import {
+    Box, Accordion, AccordionSummary, AccordionDetails, Typography, Chip
+} from '@mui/material';
+
+import useProcessVariables from '#src-app/hooks/useProcessVariables';
+import { translate } from '#src-app/hooks/useTranslations';
+import { lightTheme } from '#src-app/theme/light';
+
+import PositionedSnackbar from '../PositionedSnackbar';
+
+enum VariableTag {
+    Global = 'GlobalTag',
+    InputOutput = 'InputOutputTag',
+    ActionAssigned = 'ActionAssignedTag',
+}
+
+interface Taggable {
+    name: string,
+    tag: VariableTag,
+    color: string
+}
+
+const VariablesPanel = () => {
+    const {globalVariables, actionVariables, attendedVariables} = useProcessVariables();
+    const [expanded, setExpanded] = useState<string | null>(null);
+
+    const tagVariable = ({name, tag, color}: Taggable) => ({
+        name: `#{${name}}`,
+        tag: tag,
+        color: color
+    });
+
+    if ([...globalVariables, ...actionVariables, ...attendedVariables].length === 0) {
```

Comment:
I am Mikołaj from the future. A list of all variables is probably a bad idea, you can just check each subVariables list length. It will be probably long condition so make it separate function and name it clearly